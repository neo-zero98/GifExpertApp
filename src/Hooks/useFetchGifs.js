import { useEffect, useState } from "react"
import { getGifs } from "../Helpers/getGifs";

export const useFetchGifs = (category)=> {
    const [state, setstate] = useState({
        data: [],
        loading: true
    });


    useEffect(()=> {
        getGifs(category).then( imgs =>{
            setTimeout(()=>{
                setstate({
                    data: imgs,
                    loading: false
                });
            },3000);
        });
    },[category]);

    // setTimeout(()=>{
    //     setstate({
    //         data:[1,2,3,4,5,6,7,8,9],
    //         loading: false
    //     })
    // },3000);

    return state;
}