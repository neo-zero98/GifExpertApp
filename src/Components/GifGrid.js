import React from 'react'
import { useFetchGifs } from '../Hooks/useFetchGifs';
import GifGridItem from './GifGridItem';

const GifGrid = ({category}) => {

    const {data, loading} = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>
            {loading && <p>Loading</p>}
            <div className="card-grid animate__fadeIn">
                {
                    data.map( img => (
                        <GifGridItem 
                            key={img.id} 
                            img={img}
                        />
                    )
                )}
            </div>
        </>
    )
}

export default GifGrid;
