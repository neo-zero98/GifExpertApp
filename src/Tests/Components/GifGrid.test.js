import { shallow } from "enzyme";
import GifGrid from "../../Components/GifGrid";
import { useFetchGifs } from "../../Hooks/useFetchGifs";
jest.mock('../../Hooks/useFetchGifs');

describe('Pruebas de <GifGrid />', () => {
    const category = "Goku";

   test('toMatchSnapshot', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot(); 
   });

   test('Debe de mostrar items cuando se cargan las imagenes useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://api.yolo.jpg',
            title: 'Yolo'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot(); 
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
   });
   
    
});
