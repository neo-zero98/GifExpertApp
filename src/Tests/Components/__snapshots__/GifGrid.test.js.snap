// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifGrid /> Debe de mostrar items cuando se cargan las imagenes useFetchGifs 1`] = `
<Fragment>
  <h3>
    Goku
  </h3>
  <div
    className="card-grid animate__fadeIn"
  >
    <GifGridItem
      img={
        Object {
          "id": "ABC",
          "title": "Yolo",
          "url": "https://api.yolo.jpg",
        }
      }
      key="ABC"
    />
  </div>
</Fragment>
`;

exports[`Pruebas de <GifGrid /> toMatchSnapshot 1`] = `
<Fragment>
  <h3>
    Goku
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid animate__fadeIn"
  />
</Fragment>
`;
